src/wlapplication.cc

```diff
@@ -1066,7 +1066,8 @@ void WLApplication::warp_mouse(const Vector2i position) {
 		SDL_Window* sdl_window = g_gr->get_sdlwindow();
 		if (sdl_window != nullptr) {
 			if (!mouse_locked_) {
-				SDL_PumpEvents();
+				// Fix for #5655, TODO(klaus.halfmmann): shall we do this for macOS only?
+				NoteThreadSafeFunction::instantiate([]() { SDL_PumpEvents(); }, true);
 				SDL_FlushEvent(SDL_MOUSEMOTION);
 				SDL_WarpMouseInWindow(sdl_window, position.x, position.y);
```

Comment:
I meant not to move each call to a separate lambda but to move all three of them into the same lambda, instead of moving just the first and leaving the other two outside: 

```suggestion
				NoteThreadSafeFunction::instantiate([]() {
					SDL_PumpEvents();
					SDL_FlushEvent(SDL_MOUSEMOTION);
					SDL_WarpMouseInWindow(sdl_window, position.x, position.y);
				}, true);
```

This shouldn't cause any slowdown?